browser-test/src/header.test.ts

```diff
@@ -1,56 +1,59 @@
-import {test, expect} from '@playwright/test'
+import {test, expect} from './support/civiform_fixtures'
 import {
-  createTestContext,
   loginAsTestUser,
   validateScreenshot,
   validateAccessibility,
+  logout,
 } from './support'
 
-test.describe('Header', () => {
-  const ctx = createTestContext(/* clearDb= */ false)
-
-  test('Not logged in, guest mode enabled', async () => {
-    const {page} = ctx
-    await validateScreenshot(
-      page.getByRole('navigation'),
-      'not-logged-in-guest-mode-enabled',
-    )
-  })
-
-  // TODO(#4360): add a "Not logged in, guest mode disabled" test once we
-  // can get to the programs page without logging in, for an entity without
-  // guest mode.
-
-  test('Logged in', async () => {
-    const {page} = ctx
-    await loginAsTestUser(page)
-    await validateScreenshot(page.getByRole('navigation'), 'logged-in')
+test.describe('Header', {tag: ['@uses-fixtures']}, () => {
+  /**
+   * @todo (#4360) add a "Not logged in, guest mode disabled" test once we can get to the programs page without logging in, for an entity without guest mode.
+   */
+  test('Check screenshots and validate accessibility', async ({page}) => {
+    await test.step('Take a screenshot as a guest', async () => {
+      await logout(page)
+      await validateScreenshot(
+        page.getByRole('navigation'),
+        'not-logged-in-guest-mode-enabled',
+      )
+    })
+
+    await test.step('Take a screenshot as the test user', async () => {
+      await loginAsTestUser(page)
+      await validateScreenshot(page.getByRole('navigation'), 'logged-in')
+    })
+
+    await test.step('Passes accessibility test', async () => {
+      await validateAccessibility(page)
+    })
   })
 
-  test('Passes accessibility test', async () => {
-    const {page} = ctx
-    await validateAccessibility(page)
-  })
-
-  test('Displays the government banner', async () => {
-    const {page} = ctx
-    expect(await page.textContent('section')).toContain(
-      'This is an official government website.',
+  test('Government banner', async ({page}) => {
+    const usaBannerLocator = page.getByRole('banner').locator('.usa-banner')
```

Comment:
Technically there *should* only be a single banner role on a page. This will fail if there is more than one banner role. Some work needs to be done in the rendering side of things to make it more clear the banner (that is the first <header>)  and it's child nav have more meaningful aria settings.

Reply:
I didn't know very much about the banner role till just now.  If I'm reading these docs correctly, [docs](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/banner_role), wouldn't the parent `header` element that contains both the "This is a gov website" `section` and the `div` that represents our CF header be the one with the banner role?

Reply:
@melanie-exygy that's my understanding, but it assumes no one manually sets the role of banner on something else.